packages/date/src/index.js

```diff
@@ -653,6 +679,77 @@ export function getDateFormatsForInterval( interval, ticks = 0 ) {
 	};
 }
 
+/**
+ * Returns php date formats for the current interval.
+ * See see https://www.php.net/manual/en/datetime.format.php.
+ *
+ * @param  {string} interval Interval to get date formats for.
+ * @param  {number}    [ticks] Number of ticks the axis will have.
+ * @return {string} Current interval.
+ */
+export function getDateFormatsForIntervalPhp( interval, ticks = 0 ) {
+	let screenReaderFormat = 'F j, Y';
+	let tooltipLabelFormat = 'F j, Y';
+	let xFormat = 'Y-m-d';
+	let x2Format = 'M Y';
+	let tableFormat = defaultTableDateFormat;
+
+	switch ( interval ) {
+		case 'hour':
+			screenReaderFormat = 'gA F j, Y';
+			tooltipLabelFormat = 'gA M j, Y';
+			xFormat = 'gA';
+			x2Format = 'M j, Y';
+			tableFormat = 'h A';
+			break;
+		case 'day':
+			if ( ticks < dayTicksThreshold ) {
+				xFormat = 'j';
+			} else {
+				xFormat = 'M';
+				x2Format = 'Y';
+			}
+			break;
+		case 'week':
+			if ( ticks < weekTicksThreshold ) {
+				xFormat = 'j';
+				x2Format = 'M Y';
+			} else {
+				xFormat = 'M';
+				x2Format = 'Y';
+			}
+			screenReaderFormat = __(
+				'\\W\\e\\e\\k \\of F j, Y',
+				'woocommerce-admin'
+			);
+			tooltipLabelFormat = __(
+				'\\W\\e\\e\\k \\of F j, Y',
+				'woocommerce-admin'
+			);
```

Comment:
It seems PHP date formatting isn't ideal to be used in conjunction with words since it uses normal letters. I think this would be odd for translators and a simple mistake could break our date formatting since this string is going to be fed in our formatter.

What do you think of splitting the "Week of" string, only translate that, and add escape to all the translated characters after?

Reply:
> I think this would be odd for translators and a simple mistake could break our date formatting since this string is going to be fed in our formatter.

Yes, I had the same thought but I referred to [formatting-date-and-time document](https://wordpress.org/support/article/formatting-date-and-time/#escaping). According to the document, seems like it's okay to use escape literal characters with the translator and date formatter?

> What do you think of splitting the "Week of" string, only translate that, and add escape to all the translated characters after?

This should also work. We probably can just add escape to all "Alphabetic characters" since some translated words might become Unicode.

Either of them is fine to me. Please let me know your thoughts. Thanks.

Reply:
That's a good catch on the WP example @chihsuan.

After thinking about it more, the second approach sounds much safer still. Good idea on escaping only alphabetic characters!

Reply:
Great! Updated in e6d9ab0a0a49290e467164fcfbe4905668c2165c.